homeassistant/components/niu/binary_sensor.py

```diff
@@ -0,0 +1,93 @@
+"""Platform for binary sensor integration."""
+import logging
+
+from homeassistant.components.binary_sensor import (
+    DEVICE_CLASS_BATTERY_CHARGING,
+    DEVICE_CLASS_CONNECTIVITY,
+    DEVICE_CLASS_LOCK,
+    DEVICE_CLASS_POWER,
+    BinarySensorEntity,
+)
+
+from . import DOMAIN, NiuVehicle
+
+_LOGGER = logging.getLogger(__name__)
+
+
+SENSORS = {
+    "charging": ("Charging Status", DEVICE_CLASS_BATTERY_CHARGING, "mdi:ev-station"),
+    "connection": ("Connection Status", DEVICE_CLASS_CONNECTIVITY, "mdi:car-connected"),
+    "power": ("Power Status", DEVICE_CLASS_POWER, "mdi:power"),
+    "lock": ("Lock Status", DEVICE_CLASS_LOCK, "mdi:lock"),
```

Comment:
You don't need to set the icon here, since the logic is done in the front-end. (and will even show an open lock when it is unlocked)

https://github.com/home-assistant/frontend/blob/dev/src/common/entity/binary_sensor_icon.ts